File: SwimmingCompetitionModel/src/main/java/ro/mpp/Domain/ParticipantDTO.java
package ro.mpp.Domain;

import java.io.Serializable;
import java.util.List;
import java.util.Objects;

public class ParticipantDTO implements Serializable {
    private String nume;
    private String prenume;
    private Integer varsta;
    private List<Integer> probe;

    public ParticipantDTO(String nume, String prenume, Integer varsta, List<Integer> probe) {
        this.nume = nume;
        this.prenume = prenume;
        this.varsta = varsta;
        this.probe = probe;
    }

    public String getNume() {
        return nume;
    }

    public void setNume(String nume) {
        this.nume = nume;
    }

    public String getPrenume() {
        return prenume;
    }

    public void setPrenume(String prenume) {
        this.prenume = prenume;
    }

    public Integer getVarsta() {
        return varsta;
    }

    public void setVarsta(Integer varsta) {
        this.varsta = varsta;
    }

    public List<Integer> getProbe() {
        return probe;
    }

    public void setProbe(List<Integer> probe) {
        this.probe = probe;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        ParticipantDTO that = (ParticipantDTO) o;
        return Objects.equals(nume, that.nume) && Objects.equals(prenume, that.prenume) && Objects.equals(varsta, that.varsta) && Objects.equals(probe, that.probe);
    }

    @Override
    public int hashCode() {
        return Objects.hash(nume, prenume, varsta, probe);
    }
}

File: SwimmingCompetitionCSharpGUI/Domain/ParticipantDTO.cs
namespace SwimmingCompetition.Domain;

public class ParticipantDTO
{
    public string nume{ get; set; }
    public string prenume{ get; set; }
    public int varsta{ get; set; }
    public List<int> probe { get; set; }

    public ParticipantDTO(string nume, string prenume, int varsta, List<int> probe)
    {
        this.nume = nume;
        this.prenume = prenume;
        this.varsta = varsta;
        this.probe = probe;
    }

    protected bool Equals(ParticipantDTO other)
    {
        return nume == other.nume && prenume == other.prenume && varsta == other.varsta && probe.Equals(other.probe);
    }

    public override bool Equals(object? obj)
    {
        if (ReferenceEquals(null, obj)) return false;
        if (ReferenceEquals(this, obj)) return true;
        if (obj.GetType() != this.GetType()) return false;
        return Equals((ParticipantDTO)obj);
    }

    public override int GetHashCode()
    {
        return HashCode.Combine(nume, prenume, varsta, probe);
    }
    
}